test/common/tcp_proxy/tcp_proxy_test.cc

```diff
@@ -979,6 +979,19 @@ TEST_F(TcpProxyTest, IntermediateLogEntry) {
   filter_.reset();
 }
 
+TEST_F(TcpProxyTest, TestAccessLogOnUpstreamConnected) {
+  auto config = accessLogConfig("%UPSTREAM_HOST%");
+  config.set_flush_access_log_on_connected(true);
+
+  setup(1, config);
+  raiseEventUpstreamConnected(0);
+
+  EXPECT_EQ(access_log_data_, "127.0.0.1:80");
```

Comment:
I can't tell if this is a connection open or closed access log. Is this really the first log line?

Reply:
Yes, the fact that it is equal before calling on filter_.reset() makes sure that this was flushed on connection open. Other tests for example are  checking this only after calling filter_.reset(). I also tried seeing what happens if I don't call "config.set_flush_access_log_on_connected(true);" and that check did throw.